src/Cis194/Hw/LogAnalysis.hs

```diff
@@ -6,22 +6,44 @@ module Cis194.Hw.LogAnalysis where
 import  Cis194.Hw.Log
 
 parseMessage :: String -> LogMessage
-parseMessage s = Unknown s
+parseMessage s =
+  case words s of
+    "E":i:ws -> buildMessage (Error (read i)) ws
+    "I":ws -> buildMessage Info ws
+    "W":ws -> buildMessage Warning ws
+    _ -> Unknown s
+
+buildMessage :: MessageType -> [String] -> LogMessage
+buildMessage m (t:ws) = LogMessage m (read t) (unwords ws)
```

Comment:
If your function only makes sense for non-empty lists, it can be better to instead take the first element as a distinct argument:

```hs
buildMessage :: MessageType -> String -> [String] -> LogMessage
```

and then put the logic for what to do when you don't have one in the caller